spec/integration/apply_spec.rb

```diff
@@ -52,11 +52,8 @@ def agent_version_inventory
 
     after(:all) do
       # TODO: Extract into test helper if needed in more files
-      uri = conn_uri('ssh')
-      inventory_data = conn_inventory
-      config_data = root_config
       uninstall = '/opt/puppetlabs/bin/puppet resource package puppet-agent ensure=absent'
-      run_command(uninstall, uri, config: config_data, inventory: inventory_data)
+      run_command(uninstall, 'agent_targets', config: root_config, inventory: agent_version_inventory)
```

Comment:
I think line ~154 means we want to uninstall from everything.

Reply:
that would explain the failure in the facts module